presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -149,16 +151,20 @@ else if ((type.equals(BIGINT) || type.equals(TINYINT) || type.equals(SMALLINT) |
             ParquetIntegerStatistics parquetIntegerStatistics;
             if (statistics instanceof LongStatistics) {
                 LongStatistics longStatistics = (LongStatistics) statistics;
-                // ignore corrupted statistics
                 if (longStatistics.genericGetMin() > longStatistics.genericGetMax()) {
+                    if (failOnCorruptedParquetStatistics) {
+                        throw new ParquetCorruptionException(format("Corrupted statistics for column %s in Parquet file %s: min %d, max %d", column, id.toString(), longStatistics.genericGetMin(), longStatistics.genericGetMax()));
+                    }
                     return Domain.create(ValueSet.all(type), hasNullValue);
                 }
                 parquetIntegerStatistics = new ParquetIntegerStatistics(longStatistics.genericGetMin(), longStatistics.genericGetMax());
             }
             else {
                 IntStatistics intStatistics = (IntStatistics) statistics;
-                // ignore corrupted statistics
                 if (intStatistics.genericGetMin() > intStatistics.genericGetMax()) {
+                    if (failOnCorruptedParquetStatistics) {
```

Comment:
You can create a method like this one (feel free to find a better name) to simplify & get rid of repetition:

```
private static void failWithCorruptionException(boolean failOnCorruptedParquetStatistics, String column, ParquetDataSourceId id, Object min, Object max)
{
    if (failOnCorruptedParquetStatistics) {
        throw new ParquetCorruptionException(format("Corrupted statistics for column %s in Parquet file %s: min %s, max %s", column, id, min, max));
    }
}
```

Then calls will look like:

```
failWithCorruptionException(failOnCorruptedParquetStatistics, column, id, intStatistics.genericGetMin(), intStatistics.genericGetMax());
```